tests/integration_tests/performance/test_block_performance.py

```diff
@@ -0,0 +1,541 @@
+                                 st.function.Identity),
+        st.consumer.StatisticDef(
+            CPU_UTILIZATION_VCPUS_TOTAL, CPU_UTILIZATION_GUEST,
+            st.function.Identity)]
+
+
+def get_read_stats(env_id, fio_id, cpu):
+    """Return read-only statistics."""
+    if cpu:
+        baseline_bw_read = cpu["baseline_bw_read"][env_id][fio_id]
+        baseline_iops_read = cpu["baseline_iops_read"][env_id][fio_id]
+        baseline_slat_read = cpu["baseline_slat_read"][env_id][fio_id]
+        baseline_clat_read = cpu["baseline_clat_read"][env_id][fio_id]
+        baseline_lat_read = cpu["baseline_lat_read"][env_id][fio_id]
+        return [
+            st.consumer.StatisticDef(
+                IOPS_READ_MEAN, IOPS_READ, st.function.Avg,
+                criteria.EqualWith(baseline_iops_read["target"],
+                                   baseline_iops_read["target"] *
+                                   baseline_iops_read["delta_percentage"] /
+                                   100)),
+            st.consumer.StatisticDef(
+                IOPS_READ_STDDEV, IOPS_READ, st.function.Stddev),
+            st.consumer.StatisticDef(
+                BW_READ_MEAN, BW_READ, st.function.Avg,
+                criteria.EqualWith(baseline_bw_read["target"],
+                                   baseline_bw_read["target"] *
+                                   baseline_bw_read["delta_percentage"] /
+                                   100)),
+            st.consumer.StatisticDef(BW_READ_STDDEV, BW_READ,
+                                     st.function.Stddev),
+            st.consumer.StatisticDef(
+                SLAT_READ_MEAN, SLAT_READ, st.function.Avg,
+                criteria.EqualWith(baseline_slat_read["target"],
+                                   baseline_slat_read["target"] *
+                                   baseline_slat_read["delta_percentage"] /
+                                   100)),
+            st.consumer.StatisticDef(
+                SLAT_READ_STDDEV, SLAT_READ, st.function.Stddev),
+            st.consumer.StatisticDef(
+                CLAT_READ_MEAN, CLAT_READ, st.function.Avg,
+                criteria.EqualWith(baseline_clat_read["target"],
+                                   baseline_clat_read["target"] *
+                                   baseline_clat_read["delta_percentage"] /
+                                   100)),
+            st.consumer.StatisticDef(
+                CLAT_READ_STDDEV, CLAT_READ, st.function.Stddev),
+            st.consumer.StatisticDef(
+                LAT_READ_MEAN, LAT_READ, st.function.Avg,
+                criteria.EqualWith(
+                    baseline_lat_read["target"],
+                    baseline_lat_read["target"] *
+                    baseline_lat_read["delta_percentage"] / 100)),
+            st.consumer.StatisticDef(LAT_READ_STDDEV, LAT_READ,
+                                     st.function.Stddev)]
+    return [
```

Comment:
Can't this return list statements made generic, like a matrix where you say how do you want to calculate each param and then you just walk through the matrix and add the stuff? The code here is pretty hard to read.